client/asset/eth/nodeclient.go

```diff
@@ -174,29 +174,39 @@ func (n *nodeClient) locked() bool {
 }
 
 // transactionReceipt retrieves the transaction's receipt.
-func (n *nodeClient) transactionReceipt(ctx context.Context, txHash common.Hash) (*types.Receipt, error) {
+func (n *nodeClient) transactionReceipt(ctx context.Context, txHash common.Hash) (*types.Receipt, *types.Transaction, error) {
 	tx, blockHash, _, index, err := n.leth.ApiBackend.GetTransaction(ctx, txHash)
 	if err != nil {
 		if errors.Is(err, ethereum.NotFound) {
-			return nil, asset.CoinNotFoundError
+			return nil, nil, asset.CoinNotFoundError
 		}
-		return nil, err
+		return nil, nil, err
 	}
 	if tx == nil {
-		return nil, fmt.Errorf("%w: transaction %v not found", asset.CoinNotFoundError, txHash)
+		return nil, nil, fmt.Errorf("%w: transaction %v not found", asset.CoinNotFoundError, txHash)
 	}
 	receipts, err := n.leth.ApiBackend.GetReceipts(ctx, blockHash)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 	if len(receipts) <= int(index) {
-		return nil, fmt.Errorf("no receipt at index %d in block %s for tx %s", index, blockHash, txHash)
+		return nil, nil, fmt.Errorf("no receipt at index %d in block %s for tx %s", index, blockHash, txHash)
 	}
 	receipt := receipts[index]
 	if receipt == nil {
-		return nil, fmt.Errorf("nil receipt at index %d in block %s for tx %s", index, blockHash, txHash)
+		return nil, nil, fmt.Errorf("nil receipt at index %d in block %s for tx %s", index, blockHash, txHash)
+	}
+	return receipt, tx, nil
+}
+
+func (n *nodeClient) transactionFeesPaid(ctx context.Context, txHash common.Hash) (*big.Int, error) {
+	receipt, tx, err := n.transactionReceipt(ctx, txHash)
+	if err != nil {
+		return nil, err
 	}
-	return receipt, nil
+	hdr := n.leth.BlockChain().GetHeaderByHash(receipt.BlockHash) // n.leth.ApiBackend.HeaderByHash with no error
+	effectiveGasPrice := tx.EffectiveGasTipValue(hdr.BaseFee)
```

Comment:
@buck54321 pointed this out, but I think what I suggested was incomplete.  What I was basing this on was https://github.com/ethereum/go-ethereum/blob/aa69d36152becdc819f323b5ae05e249d984d609/internal/ethapi/api.go#L1572-L1573
`gasPrice := new(big.Int).Add(header.BaseFee, tx.EffectiveGasTipValue(header.BaseFee))`

Reply:
Ok. I also took out a check in this function for the tx status. I think that some gas may be used regardless if it is mined so couldn't think of a good reason to fail if status is not OK. Maybe it was just a copy pasta or if there was a reason can add back.